File: 121.best-time-to-buy-and-sell-stock.js
/*
 * @lc app=leetcode id=121 lang=javascript
 *
 * [121] Best Time to Buy and Sell Stock
 */

// @lc code=start
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let max = 0
    for (let i = 0; i < prices.length; i++) {
        let curr = prices[i]
        let maxFutures = Math.max(...prices.splice(i + 1))
        let profit = maxFutures - curr
        console.log(profit)
        if (profit > max) max = profit
    }
    return max
};
// @lc code=end

File: 121.best-time-to-buy-and-sell-stock.py
#
# @lc app=leetcode id=121 lang=python
#
# [121] Best Time to Buy and Sell Stock
#

# @lc code=start
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) == 0:
            return 0
        min_price = prices[0]
        max_profit = 0
        for i in range(1, len(prices)):
            if prices[i] < min_price:
                min_price = prices[i]
            elif prices[i] - min_price > max_profit:
                max_profit = prices[i] - min_price
        return max_profit
# @lc code=end
